docs/Current_Code.py

```diff
@@ -0,0 +1,118 @@
+                print(nums)
+                if (len(nums) == 2):
+                    return nums
+
+    def readData(self,x):
+        # Plotting array
+        ax = []
+        ay1 = []  # Define y value for Current
+        ay2 = []  # Define y value for Voltage
+        CurrentArray = []
+        VoltageArray = []
+        start = time.time()
+        lastHigh = time.time()
+        for i in range(0, int(x)):
+            last = time.time()
+            self.analog_d9p.write(i / 255)  # 100 % duty cycle (2.5V analog value)
+            val0 = 0 # Voltage
+            val2 = 0 # Current
+            nums = self.readPin()
+            val2 = nums[1]
+            val0 = nums[0]
+            CurrentArray.append(val2)
+            VoltageArray.append(val0)
+            time.sleep(0.001)
+            end = time.time()  # clock for the real time plotting
+            xl = end - start
+            # --------Plotting--------------
+            ax.append(round(xl,4))  # real time on x-axis
+            ay1.append(round(val2,4))  # assign Current with respect to xl
+            ay2.append(round(val0,4))  # assign Voltage with respect to xl
+        timeArray = numpy.asarray(ax)
+        CurrentArray = numpy.asarray(ay1)
+        VoltageArray = numpy.asarray(ay2)
+        numpy.savetxt('dataset.csv',numpy.column_stack((timeArray,CurrentArray,VoltageArray)))
+        return (ax,ay1,ay2)
+
+    def plotData(self,x):
+        # Plotting array
+        ax = []
+        ay1 = []  # Define y value for Current
+        ay2 = []  # Define y value for Voltage
+        CurrentArray = []
+        VoltageArray = []
+        plt.ion()  # Open a plotting window
+        start = time.time()
+        for i in range(0, int(x)):
+            self.analog_d9p.write(i / 255)  # 100 % duty cycle (2.5V analog value)
+            nums = self.readPin()
+            val2 = nums[1] # Current
+            val0 = nums[0] # Voltage
+            CurrentArray.append(val2)
+            VoltageArray.append(val0)
+            time.sleep(0.001)
+            end = time.time()  # clock for the real time plotting
+            xl = end - start
+            # --------Plotting--------------
+            ax.append(xl)  # real time on x-axis
+            ay1.append(val2)  # assign Current with respect to xl
+            ay2.append(val0)  # assign Voltage with respect to xl
+            plt.clf()  # clear previous plots
+
+            plt.xlim(xl - 40, xl + 40)  # Set up x range
+            plt.ylim(-2500, 2500)  # Set up y range
+
+            plt.plot(ax, ay1, color='blue')
+            plt.plot(ax, ay2, color='red')
+
+            plt.xlabel('Time')  # name x-axis
+            plt.ylabel('Current/Voltage')  # name y-axis
+
+            plt.pause(0.00001)  # pause for 0.01 second
+            plt.savefig('./testloop1.png')  # Save the poltting as a picture
+            plt.show()
+        plt.pause(10)
+        return (ax,ay1,ay2)
+
+if __name__ == '__main__':
+    test = myArduinoBoard()
```

Comment:
we usually do testing in the test directory.  maybe a better name needed here.  I might think of it this way.  Call the class ``ArduinoBoard`` because it describes an arduino board.  Call your instance of that class myArduinoBoard, so
```
if __name__ == '__main__':
    my_arduino_board = ArduinoBoard()
    x = 255
    my_arduino_board.readData(x)
    my_arduino_board.plotData(x)
```